### PR TITLE
[Feature]Add augments to models/utils

### PR DESCRIPTION
**Pull request by HIT-cwh**
Hi,
A new class named Augments is added in this pr. 
Example:
You can use cutmix like this:
```
augments_cfg = dict(type='BatchCutMix', alpha=1., num_classes=10, prob=1.)
augments = Augments(augments_cfg)
```
And you can use both cutmix and mixup like this:

```
augments_cfg = [
    dict(type='BatchCutMix', alpha=1., num_classes=10, prob=1.),
    dict(type='BatchMixup', alpha=1., num_classes=10, prob=0.6),
    dict(type='Identity', num_classes=10, prob=0.4)
]
augments = Augments(augments_cfg)
```
    
To decide which augmentation within OneOf block is used the following rule is applied.
    We normalize all probabilities within augments_cfg to one. After this we pick augmentation based on the normalized probabilities. In the example above BatchCutMix has probability 1.0, BatchMixup probability 0.6 and Identity probability 0.4. After normalization, they become 0.5, 0.3 and 0.2. Which means that we decide if we should use BatchCutMix with probability 0.5, BatchMixup 0.3 and Identity otherwise 0.2.

Pls kindly take a look.

### CODE REVIEW

File: mmcls/models/utils/augment/cutmix.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    def rand_bbox_minmax(self, img_shape, minmax, count=None):
+        """Min-Max CutMix bounding-box Inspired by Darknet cutmix impl,
+        generates a random rectangular bbox based on min/max percent values
+        applied to each dimension of the input image.
+
+        Typical defaults for minmax are usually in the  .2-.3 for min and
+        .8-.9 range for max.
+
+        Args:
+            img_shape (tuple): Image shape as tuple
```
Comment (ycxioooong) [inline mmcls/models/utils/augment/cutmix.py:51]: Can we restrict the tuple length here? i.e., len(img_shape) == 2. And inform that the first number stands for height and the second one is width.

File: mmcls/models/utils/augment/cutmix.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        cut_w = np.random.randint(
+            int(img_w * minmax[0]), int(img_w * minmax[1]), size=count)
+        yl = np.random.randint(0, img_h - cut_h, size=count)
+        xl = np.random.randint(0, img_w - cut_w, size=count)
+        yu = yl + cut_h
+        xu = xl + cut_w
+        return yl, yu, xl, xu
+
+    def rand_bbox(self, img_shape, lam, margin=0., count=None):
+        """Standard CutMix bounding-box Generates a random square bbox based on
```
Comment (ycxioooong) [inline mmcls/models/utils/augment/cutmix.py:69]: Generates -> that generates

File: mmcls/models/utils/augment/cutmix.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        lambda value. This impl includes support for enforcing a border margin
```
Comment (ycxioooong) [inline mmcls/models/utils/augment/cutmix.py:70]: impl -> implementation

File: mmcls/models/utils/augment/cutmix.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        return yl, yh, xl, xh
+
+    def cutmix_bbox_and_lam(self,
+                            img_shape,
+                            lam,
+                            ratio_minmax=None,
+                            correct_lam=True,
+                            count=None):
+        """Generate bbox and apply lambda correction."""
+        if ratio_minmax is not None:
```
Comment (ycxioooong) [inline mmcls/models/utils/augment/cutmix.py:99]: missing docstring here

File: mmcls/models/utils/augment/cutmix.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+@AUGMENT.register_module(name='BatchCutMix')
+class BatchCutMixLayer(BaseCutMixLayer):
+    """CutMix layer for batch CutMix."""
+
+    def __init__(self, *args, **kwargs):
+        super(BatchCutMixLayer, self).__init__(*args, **kwargs)
+
+    def cutmix(self, img, gt_label):
+        one_hot_gt_label = F.one_hot(gt_label, num_classes=self.num_classes)
+        # r = np.random.rand(1)
```
Comment (ycxioooong) [inline mmcls/models/utils/augment/cutmix.py:123]: remember to delete these comments

File: mmcls/models/utils/augment/identity.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import torch.nn.functional as F
+
+from .builder import AUGMENT
+
+
+@AUGMENT.register_module(name='Identity')
+class Identity(object):
+    """Change gt_label to one_hot encoding and keep img the same.
```
Comment (ycxioooong) [inline mmcls/models/utils/augment/identity.py:8]: keep img as the same

File: mmcls/models/utils/augment/augments.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
+import random
+
+import numpy as np
+
+from .builder import build_augment
+
+
+class Augments:
+    """Data augments.
```
[the remark this answers is not in the document]
Comment (HIT-cwh) [reply mmcls/models/utils/augment/augments.py:9]: class Augments is more than intra-batch data augmentations, but can be other kinds of data augmentations, such as mix_elem, mix_pair and something like that.

File: mmcls/models/utils/augment/cutmix.py (hunk 1 above)
Comment (HIT-cwh) [reply mmcls/models/utils/augment/cutmix.py:51]: imgs in pytorch network are  in BCHW format, so , perhaps, we don't have to restrict the tuple length here.

**Review by ycxioooong — COMMENTED** [review]
Almost LGTM now.

**Comment by ycxioooong** [comment]
Please resolve the conflict @HIT-cwh

**Comment by ycxioooong** [comment]
Hi @HIT-cwh, the linting failed, please resolve this issue at your working time. Thanks.

**Review by ycxioooong — APPROVED** [review]
LGTM now

**Comment by mzr1996** [comment]
> Hi,
> A new class named Augments is added in this pr.
> Example:
> You can use cutmix like this:
> 
> ```
> augments_cfg = dict(type='BatchCutMix', alpha=1., num_classes=10, prob=1.)
> augments = Augments(augments_cfg)
> ```
> 
> And you can use both cutmix and mixup like this:
> 
> ```
> augments_cfg = [
>     dict(type='BatchCutMix', alpha=1., num_classes=10, prob=1.),
>     dict(type='BatchMixup', alpha=1., num_classes=10, prob=0.6),
>     dict(type='Identity', num_classes=10, prob=0.4)
> ]
> augments = Augments(augments_cfg)
> ```
> 
> To decide which augmentation within OneOf block is used the following rule is applied.
> We normalize all probabilities within augments_cfg to one. After this we pick augmentation based on the normalized probabilities. In the example above BatchCutMix has probability 1.0, BatchMixup probability 0.6 and Identity probability 0.4. After normalization, they become 0.5, 0.3 and 0.2. Which means that we decide if we should use BatchCutMix with probability 0.5, BatchMixup 0.3 and Identity otherwise 0.2.
> 
> Pls kindly take a look.

Hello, maybe you need to update this use case.

**Comment by HIT-cwh** [comment]
> > Hi,
> > A new class named Augments is added in this pr.
> > Example:
> > You can use cutmix like this:
> > ```
> > augments_cfg = dict(type='BatchCutMix', alpha=1., num_classes=10, prob=1.)
> > augments = Augments(augments_cfg)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > And you can use both cutmix and mixup like this:
> > ```
> > augments_cfg = [
> >     dict(type='BatchCutMix', alpha=1., num_classes=10, prob=1.),
> >     dict(type='BatchMixup', alpha=1., num_classes=10, prob=0.6),
> >     dict(type='Identity', num_classes=10, prob=0.4)
> > ]
> > augments = Augments(augments_cfg)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > To decide which augmentation within OneOf block is used the following rule is applied.
> > We normalize all probabilities within augments_cfg to one. After this we pick augmentation based on the normalized probabilities. In the example above BatchCutMix has probability 1.0, BatchMixup probability 0.6 and Identity probability 0.4. After normalization, they become 0.5, 0.3 and 0.2. Which means that we decide if we should use BatchCutMix with probability 0.5, BatchMixup 0.3 and Identity otherwise 0.2.
> > Pls kindly take a look.
> 
> Hello, maybe you need to update this use case.

Fixed.
